addons/popochiu/editor/main_dock/popochiu_row/popochiu_row.gd

```diff
@@ -0,0 +1,111 @@
+@tool
+extends HBoxContainer
+## The row that is created for Rooms, Characters, Inventory items, Dialogs, Props, Hotspots, Regions,
+## Walkable areas, and Markers in the dock.
+
+signal clicked(node: HBoxContainer)
+
+enum MenuOptions {
+	DELETE
+}
+
+const SELECTED_FONT_COLOR := Color("706deb")
+const AudioCue := preload("res://addons/popochiu/engine/audio_manager/audio_cue.gd")
+
+var path := ""
+var is_menu_hidden := false
+var type := -1
+
+var _delete_dialog: PopochiuEditorHelper.DeleteConfirmation = null
+
+@onready var label: Label = %Label
+@onready var tag: TextureRect = %Tag
+@onready var btn_menu: MenuButton = %BtnMenu
+@onready var menu_popup: PopupMenu = btn_menu.get_popup()
+@onready var buttons_container: HBoxContainer = %ButtonsContainer
+@onready var dflt_font_color: Color = label.get_theme_color("font_color")
+
+
+#region Godot ######################################################################################
+func _ready() -> void:
+	label.text = str(name)
+	tooltip_text = path
+	
+	# Assign icons
+	btn_menu.icon = get_theme_icon("GuiTabMenuHl", "EditorIcons")
+	
+	# Create the context menu based checked the type of Object this row represents
+	_create_menu()
+	
+	tag.hide()
+	if is_menu_hidden:
+		btn_menu.hide()
+	
+	gui_input.connect(_check_click)
+	menu_popup.id_pressed.connect(_menu_item_pressed)
+
+
+#endregion
+
+#region Virtual ####################################################################################
+func _remove_object() -> void:
+	pass
+
+
+#endregion
+
+#region Public #####################################################################################
+func select() -> void:
+	label.add_theme_color_override("font_color", SELECTED_FONT_COLOR)
+	clicked.emit(self)
+
+
+func deselect() -> void:
+	label.add_theme_color_override("font_color", dflt_font_color)
+
+
+func remove_menu_option(opt: int) -> void:
+	menu_popup.remove_item(menu_popup.get_item_index(opt))
+
+
+func add_button(btn: Button) -> void:
+	buttons_container.add_child(btn)
+
+
+#endregion
+
+#region Private ####################################################################################
+func _create_menu() -> void:
+	menu_popup.clear()
+	
+	for option in _get_menu_cfg():
+		if option == null:
```

Comment:
I know this put a bit of a burden, but I would change this `null` or a constant like OBJ_ROW_MENU_SEPARATOR

Reply:
Sounds good. I'll make the change.

Reply:
Done.